[BUG] When running extractor for OAS version 3 Json it is converting boolean values to string

### Release version

4.0.3

### Describe the bug

When running extractor pipeline with option OAS v3 json, specification.json generated will convert the boolean values to string which causing linting failed. 

### Expected behavior

boolean values should not converted to string. 

example: expected result
"required" : true

generated specification result:
"required" : "true"


### Actual behavior

generated specification result:
"required" : "true"
`"delete": {
        "tags": [
          "OptIn"
        ],
        "summary": "Delete Opt-In",
        "description": "Delete Opt-In status for a user. Resets them to not opted in.",
        "operationId": "delete-users-userid",
        "parameters": [
          {
            "name": "userId",
            "in": "path",
            **

> "required": "true",

**
            "schema": {
              "type": "string"
            }
          }
        ],
        "responses": {
          "200": {
            "description": "Deleted Opt-In Data",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/OptInData"
                },
                "example": {
                  "userId": "string"
                }
              }
            }
          },`

### Reproduction Steps

Run extractor pipeline with the generator option OpenAPIV3Json and analyse the specification.json. 

`83:11    error  path-params               Path parameter "userId" must have "required" property that is set to "true".                                                                                    paths./AccountLinks/{userId}/accounts/{accountNumber}.get.parameters[0]
  87:25    error  oas3-schema               "required" property must be equal to one of the allowed values.                                                                                                 paths./AccountLinks/{userId}/accounts/{accountNumber}.get.parameters[0].required
  87:25    error  oas3-schema               "required" property type must be boolean.                                                                                                                       paths./AccountLinks/{userId}/accounts/{accountNumber}.get.parameters[0].required`

--- a/tools/code/extractor/Api.cs
+++ b/tools/code/extractor/Api.cs
@@ -141,10 +141,11 @@ internal static class Api
         {
             DefaultApiSpecification.Wadl => "wadl-link",
             DefaultApiSpecification.OpenApi openApi =>
-                openApi.Version switch
+                (openApi.Version, openApi.Format) switch
                 {
-                    OpenApiSpecVersion.OpenApi2_0 => "swagger-link",
-                    OpenApiSpecVersion.OpenApi3_0 => "openapi-link",
+                    (OpenApiSpecVersion.OpenApi2_0, _) => "swagger-link",
+                    (OpenApiSpecVersion.OpenApi3_0, OpenApiFormat.Yaml) => "openapi-link",
+                    (OpenApiSpecVersion.OpenApi3_0, OpenApiFormat.Json) => "openapi+json-link",
                     _ => throw new NotSupportedException()
                 },
             _ => throw new NotSupportedException()
@@ -155,13 +156,6 @@ internal static class Api
         logger.LogInformation("Writing API specification file {filePath}...", specificationFile.Path);
         switch (defaultSpecification)
         {
-            // APIM exports OpenApiv3 to YAML. Convert to JSON if needed.
-            case DefaultApiSpecification.OpenApi openApi when openApi.Version is OpenApiSpecVersion.OpenApi3_0 && openApi.Format is OpenApiFormat.Json:
-                {
-                    var bytes = ConvertYamlStreamToJson(downloadFileStream);
-                    await specificationFile.OverwriteWithBytes(bytes, cancellationToken);
-                    break;
-                }
             // APIM exports OpenApiv2 to JSON. Convert to YAML if needed.
             case DefaultApiSpecification.OpenApi openApi when openApi.Version is OpenApiSpecVersion.OpenApi2_0 && openApi.Format is OpenApiFormat.Yaml:
                 {
@@ -173,12 +167,6 @@ internal static class Api
                 await specificationFile.OverwriteWithStream(downloadFileStream, cancellationToken);
                 break;
         }
-    }
-
-    private static byte[] ConvertYamlStreamToJson(Stream stream)
-    {
-        var yaml = ConvertStreamToYamlObject(stream);
-        return JsonSerializer.SerializeToUtf8Bytes(yaml, new JsonSerializerOptions { WriteIndented = true });
     }
 
     private static object ConvertStreamToYamlObject(Stream stream)

